modules/MarketAnalysis.py

```diff
@@ -148,3 +196,84 @@ def get_percentile(self, rates, lending_style, use_numpy=use_numpy):
             result = self.percentile(sorted(rates), lending_style / 100.0)
         result = truncate(result, 6)
         return result
+
+    def get_golden_cross_rate(self, cur, rates_df, short_period=150, long_period=1800):
+        # TODO These don't need to be rolling mean, simple mean on the last X will do (and will be faster)
+        short_rate = pd.rolling_mean(rates_df[1], window=short_period, min_periods=1).iloc[-1]
+        long_rate = pd.rolling_mean(rates_df[1], window=long_period, min_periods=1).iloc[-1]
+        # TODO remove the sys writes
+        if short_rate > long_rate:
+            sys.stdout.write("Short higher : ")
+            rate = short_rate if rates_df.iloc[-1][1] < short_rate else rates_df.iloc[-1][1]
+        else:
+            sys.stdout.write("Long  higher : ")
+            rate = long_rate
+        # TODO Needs config option
+        rate = rate * 1.05
+        return rate
+
+    def create_connection(self, db_path='/tmp/plb.db', db_type='sqlite3'):
```

Comment:
Yea, OK. Good idea. 

I'll put it in with my next commit. Wasn't really expecting anyone to look at it yet, so I didn't bother with a lot of stuff :)